### PR TITLE
reduce header file to boost cpp_wrapper build.

### PR DESCRIPTION
**Pull request by xuhancn**
1. Reduce cpp_wrapper un-used header files.
2. Clean pch cache, when use_pch is False.

The first change will reduce the build time from 7.35s to 4.94s.

Before change:
image
After change:
image

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

File: torch/extension.h [hunk 1]
```diff
@@ -1,6 +1,9 @@
 #pragma once
 
+#ifndef CPP_WRAPPER_MODULE
```
Comment (ezyang) [inline torch/extension.h:3]: Can you namespace this macro

**Review by ezyang — APPROVED** [review]

File: torch/extension.h (hunk 1 above)
Comment (xuhancn) [reply torch/extension.h:3]: @ezyang  use "INDUCTOR_CPP_WRAPPER" ?

Comment (ezyang) [reply torch/extension.h]: sure, or maybe TORCH

**Review by jansel — APPROVED** [review]

File: torch/extension.h [hunk 2]
```diff
@@ -1,6 +1,9 @@
 #pragma once
 
+#ifndef INDUCTOR_CPP_WRAPPER
```
Comment (jgong5) [inline torch/extension.h:3]: Suggest to use `TORCH_INDUCTOR_CPP_WRAPPER` per suggested by @ezyang

**Review by jgong5 — APPROVED** [review]

File: torch/extension.h (hunk 2 above)
Comment (xuhancn) [reply torch/extension.h]: Done

**Pull request closed without merging**